src/v/cluster/scheduling/partition_allocator.cc

```diff
@@ -321,7 +321,8 @@ result<allocated_partition> partition_allocator::reallocate_partition(
   partition_constraints p_constraints,
   const partition_assignment& current_assignment,
   const partition_allocation_domain domain,
-  const std::vector<model::node_id>& replicas_to_reallocate) {
+  const std::vector<model::node_id>& replicas_to_reallocate,
+  node2count_t* existing_replica_counts) {
```

Comment:
nit: I was debating pointer vs an optional<reference_wrapper<node2count_t>>.. the latter looks really ugly I guess, sigh

Reply:
Yeah, I always feel a bit guilty about using raw pointers... OTOH I am always drawn to it by its simplicity and versatility :)

Reply:
@bharathv @ztlpn https://github.com/redpanda-data/redpanda/blob/dev/src/v/net/transport.h#L111 oh I saw this pattern being discussed on slack and tried it. passed code review :P